Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -304,25 +304,23 @@ struct OrdersUpsertUseCase {
     /// Updates, inserts, or prunes the provided `storageOrder`'s custom fields using the provided `readOnlyOrder`'s custom fields
     ///
     private func handleOrderCustomFields(_ readOnlyOrder: Networking.Order, _ storageOrder: Storage.Order, _ storage: StorageType) {
-        // Upsert the `customFields` from the `readOnlyOrder`
-        readOnlyOrder.customFields.forEach { readOnlyCustomField in
-            if let existingStorageMetaData = storage.loadOrderMetaData(siteID: readOnlyOrder.siteID,
-                                                                       orderID: storageOrder.orderID,
-                                                                       metadataID: readOnlyCustomField.metadataID) {
-                existingStorageMetaData.update(with: readOnlyCustomField)
-            } else {
-                let newStorageMetaData = storage.insertNewObject(ofType: Storage.MetaData.self)
-                newStorageMetaData.update(with: readOnlyCustomField)
-                storageOrder.addToCustomFields(newStorageMetaData)
-            }
-        }
 
-        // Now, remove any objects that exist in `storageOrder.customFields` but not in `readOnlyOrder.customFields`
+        // Remove any objects that exist in `storageOrder.customFields` regarding a specific order.
         storageOrder.customFields?.forEach { storageCustomField in
-            if readOnlyOrder.customFields.first(where: { $0.metadataID == storageCustomField.metadataID } ) == nil {
-                storageOrder.removeFromCustomFields(storageCustomField)
-                storage.deleteObject(storageCustomField)
-            }
+            storageOrder.removeFromCustomFields(storageCustomField)
+            storage.deleteObject(storageCustomField)
+        }
+
+        // Create `customFields` objects from the `readOnlyOrder`
+        let newStorageMetaDataArray = readOnlyOrder.customFields.map { readOnlyCustomField in
+            let newStorageMetaData = storage.insertNewObject(ofType: Storage.MetaData.self)
+            newStorageMetaData.update(with: readOnlyCustomField)
+            return newStorageMetaData
+        }
+
+        // Batch writing process of multiple custom fields
+        if !newStorageMetaDataArray.isEmpty {
+            storageOrder.addToCustomFields(NSSet(array: newStorageMetaDataArray))
```

Comment:
@pmusolino Just sharing an observation from some debugging - this line caused the upsert to be triggered twice, with `newStorageMetaDataArray` containing a different custom field object (different Core Data object ID) each time. It's the expected result of the updated approach in this PR, with pre-existing custom fields being deleted and new custom fields re-added.

Reply:
Thanks, Jaclyn!

From the comment of [the failing unit test](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Yosemite/YosemiteTests/Stores/OrderStoreTests.swift#L971) I wonder whether the updated approach of deleting all the stored custom fields before adding new custom fields will cause any reload or UI issues in screens where we use `EntityListener` to listen to changes to product. 🤔